pkg/blockchainupdates/nuts_publisher.go

```diff
@@ -0,0 +1,63 @@
+package blockchainupdates
+
+import (
+	"context"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+	"log"
+)
+
+type BUpdatesInfo struct {
+	Height      uint64
+	VRF         proto.B58Bytes
+	BlockID     proto.BlockID
+	BlockHeader *proto.BlockHeader
+}
+
+type BUpdatesExtensionState struct {
+	currentState  *BUpdatesInfo
+	previousState *BUpdatesInfo // this information is what was just published
+}
+
+func NewBUpdatesExtensionState() *BUpdatesExtensionState {
+	return &BUpdatesExtensionState{}
+}
+
+func (bu *BUpdatesExtensionState) hasStateChanged() {
+
+}
+
+func (bu *BUpdatesExtensionState) RunBlockchainUpdatesPublisher(ctx context.Context, updatesChannel <-chan BUpdatesInfo) {
+	nc, err := nats.Connect(nats.DefaultURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer nc.Close()
+
+	func(ctx context.Context, updatesChannel <-chan BUpdatesInfo) {
```

Comment:
I think you've forgotten to add `go` word before the call.

Reply:
Could be back then! Now it's a runPublisher function that shouldn't be a goroutine, otherwise the defered nc.Close() will be triggered
![Screenshot from 2024-11-20 02-44-03](https://github.com/user-attachments/assets/4057c4c7-10ff-4b73-898d-3841d8d7c94a)